src/wallet/rpcwallet.cpp

```diff
@@ -3357,7 +3399,7 @@ UniValue z_listreceivedbyaddress(const UniValue& params, bool fHelp)
     UniValue result(UniValue::VARR);
     std::vector<SproutNoteEntry> sproutEntries;
     std::vector<SaplingNoteEntry> saplingEntries;
-    pwalletMain->GetFilteredNotes(sproutEntries, saplingEntries, fromaddress, nMinDepth, false, false);
+    pwalletMain->GetFilteredNotes(sproutEntries, saplingEntries, fromaddress, nMinDepth, false, false, timestamp);
```

Comment:
I think that instead of calling `GetFilteredNotes()` you should call `GetNotesByType` in two blocks below like:
```
if (boost::get<libzcash::SproutPaymentAddress>(&zaddr) != nullptr) {
```

This will also allow you to get rid of `getPaginatedEntries()` function which I suspect has a problem with overflow anyway.

Please remember to pass `limit` and `offset`when calling ``GetNotesByType`

Reply:
Yea, i am on that part now.